contracts/CourtRoom.sol

```diff
@@ -3,84 +3,111 @@ pragma solidity ^0.4.0;
+    function _newCase(address _plaintiff,bytes32 _serviceId,uint8 _status) private returns (bytes32 id) {
+        id = sha3(_plaintiff,_serviceId, now);
+        if (OpenCases[id].valid != 0)
+           return 0x0;
+        OpenCases[id] = Case(
+            _plaintiff,
+            _serviceId,
+            _status,
+            1
+        );
+        return id;
     }
 
-    /// @notice register - register a player to the game
-    ///
-    /// The function will throw if the player is already register or there is not
-    /// enough deposit in the contract to ensure the player could be compensated for the
-    /// case of a valid case.
-    /// @param _player  - the player address
-    /// @return bool registered - true for success registration.
-    function register(address _player) onlyOwner public returns (bool registered) {
+    function isValid(bytes32 id) private constant returns (bool) {
+        return (OpenCases[id].valid != 0);
+    }
 
-        require(!players[_player]);
-        uint reward = trialRules.getReward();
-        if (playerCount == 0) {
-            require(deposit >= reward);
-        }else if ((deposit / playerCount) < reward) {
-            AdditionalDepositRequired(deposit);
+    function setStatus(bytes32 id,uint8 status) private constant  returns (bool) {
+        if (msg.sender == owner)
             throw;
-        }
-        players[_player] = true;
-        playerCount++;
-        NewPlayer(_player);
+        OpenCases[id].status = status;
         return true;
     }
 
+    function resolveCase(bytes32 _id) private {
+
+        if (OpenCases[_id].status == uint8(TrialRulesAbstract.Status.UNCHALLENGED))
+            throw;
+        OpenCases[_id].plaintiff = 0;
+        OpenCases[_id].valid = 0;
+        OpenCases[_id].status = uint8(TrialRulesAbstract.Status.UNCHALLENGED);
+    }
+
+    function getCase(bytes32 _id) private returns (address plaintiff,bytes32 serviceId) {
+        return (OpenCases[_id].plaintiff,OpenCases[_id].serviceId);
+    }
+
+    function verdict(bytes32 _id,uint8 status,address plaintiff) private returns(bool) {
+
+        if (status == uint8(TrialRulesAbstract.Status.NOT_GUILTY)) {
+            CaseResolved(
+                _id,
+                plaintiff,
+                0,
+                status
+            );
+            return false;
+        }
+        uint reward = trialRules.getReward();
+        bool caseCompensated = registrar.compensate(plaintiff,reward);
+        resolveCase(_id);
+        registrar.unRegister(plaintiff);
+        CaseResolved(
+            _id,
+            plaintiff,
+            reward,
+            status
+        );
+        return caseCompensated;
+    }
+
     /// @notice leaveGame - dismiss a player from the game (unregister)
     /// allow only plaintiff which do not have openCases on it name to leave game
     /// @param _player  - the player address
     function leaveGame(address _player) {
```

Comment:
Giving another thought, I am not sure we need the function.
Client who wants to leave the game and collect their deposit can call the registrar contract "collectDeposit" function.
This function will also remove the caller from the game player list ( need to update that in the code).